.github/workflows/ci-build.yaml

```diff
@@ -0,0 +1,98 @@
+name: Continuous integration
+
+on:
+  # Trigger the workflow on push or pull request,
+  # but only for the master branch
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  lint_and_typecheck:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.7]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r build/test-requirements.txt
+    - name: Lint with flake8
+      run: |
+        flake8 .
+    - name: Type check with mypy
+      run: |
+        mypy jax
+
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        include:
+          - python-version: 3.6
+            os: macos-latest
+            enable-x64: disable-x64
+            num_generated_cases: 10
+          - python-version: 3.7
+            os: ubuntu-latest
+            enable-x64: disable-x64
+            num_generated_cases: 25
+          - python-version: 3.7
+            os: ubuntu-latest
+            enable-x64: enable-x64
+            num_generated_cases: 25
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r build/test-requirements.txt
+    - name: Run tests
+      run: |
+        pip install -e .
+        [ ${{ matrix.enable-x64 }} = enable-x64 ] && JAX_ENABLE_X64=1 || JAX_ENABLE_X64=0 
+        export JAX_NUM_GENERATED_CASES=${{ matrix.num_generated_cases }}
```

Comment:
FYI you can set env vars directly for the step: https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsenv (this is good too though)

Reply:
Done for `JAX_NUM_GENERATED_CASES`. I don't know how to do this for `JAX_ENABLE_X64` because I want a meaningful string to show up in the job name (not just `0` and `1`.)

Reply:
If you want, apparently you can use operators like `==` in `${{ }}` expressions: https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#operators

Reply:
I looked at that. Unfortunately there doesn't seem to be an easy way to cast a `bool` to an integer. And there's no ternary conditional operator.